scripts/variantstore/wdl/ngs_filter_extract.wdl

```diff
@@ -9,14 +9,19 @@ workflow NgsFilterExtract {
         File reference_index
         File reference_dict
 
-        String fq_sample_table
-        String fq_alt_allele_table
-        String query_project
+
+        String data_project
+        String? default_dataset
+
+        String query_project = data_project
+
+        String fq_sample_table = "~{data_project}.~{default_dataset}.sample_info"
```

Comment:
what happens here if default_dataset is not defined?

Reply:
Yeah I struggled with this a little bit to have a principle of least surprise...

If you don't supply a default_dataset AND you don't override these parameters... things won't work (ie the dataset reference won't be valid).  Not much we can do there since there isn't great parameter validation in WDL with xor or groups of settings, etc.

We could make "default_dataset" required.  That might make more sense since the vast majority of times... you mean to set that and having the validation would be nice.  And in the rare case where someone wants to override all the table parameters... they'll still have to specify the default even though technically it's not needed.  But that's the exceptional case... wdyt?

Reply:
After the slack convo -- I think `default_dataset` should be non-optional.

Reply:
I have a slight preference for making `default_dataset` required. Like you said, if it's going to be used the vast majority of the time, having it in the "required" section of inputs is going to lead to better outcomes than if it's a random optional input that really should most of the time be defined. 

In the rarer cases where someone provides the fq table inputs, default_dataset will just get ignored, right?